### PR TITLE
VSI submission: redirect terminal input through socket (1.0.2)

### PR DESCRIPTION
**Pull request by levitte**
Same as #1575, but for 1.0.2

### CODE REVIEW

File: apps/s_server.c [hunk 1]
```diff
@@ -2802,8 +2839,20 @@ static int www_body(char *hostname, int s, int stype, unsigned char *context)
...
+        /* For 64-bit --> 32-bit restricted APIs (IOCTL) */
+#  if __INITIAL_POINTER_SIZE == 64
+#   pragma __required_pointer_size __save
+#   pragma __required_pointer_size 32
+#  endif
+        unsigned int sl ;
+#  if __INITIAL_POINTER_SIZE == 64
+#   pragma __required_pointer_size __restore
+#  endif
+# else
```
Comment (dot-asm) [inline apps/s_server.c]: I fail to grasp this. Why the "dance" with pointers? Doesn't BIO_socket_ioctl take care of if? How come long works for all, but not VMS? And why is it uninitialized on VMS?

Comment (levitte) [inline apps/s_server.c]: It gets removed in 3065ca9647a7d6435d8174662cd32c52eb75d852

Comment (levitte) [inline apps/s_server.c]: Never mind my previous comment.  I'll have that removed

**Comment by dot-asm** [comment]
Doesn't one need to modify build script too? I mean where does it compile vms_term_sock.c?

**Comment by levitte** [comment]
Erroneous pointer_size guard gone.
vms_term_sock added to build script

**Comment by dot-asm** [comment]
There is certain ambiguity in #1575. There was suggestion and you said you'll look into it. Does it mean that there will be changes? Meanwhile I'm signing off for today...

**Comment by levitte** [comment]
Whatever changes made in #1575 will be ported here as well...  or really, it's the other way around, as some changes from this branch aren't needed in master.

**Comment by levitte** [comment]
I'll get to that tomorrow.  Way too late to do that now.

**Comment by levitte** [comment]
Ok, I did a little bit of refactoring here.  Please check it out.

**Comment by richsalz** [comment]
+1. nice.

File: apps/apps.c [hunk 2]
```diff
@@ -3198,6 +3198,32 @@ int app_isdir(const char *name)
 #endif
 
 /* raw_read|write section */
+#if defined(__VMS)
+# include "vms_term_sock.h"
+static int stdin_sock = -1;
+static void close_stdin_sock(void)
```
Comment (dot-asm) [inline apps/apps.c]: Doesn't style guide suggest empty line prior function? This is a side note, i.e. it's not necessarily to act upon it, you are free to use the plus-one request has already.

**Comment by levitte** [comment]
Added a couple of fixups.  Two bugs and the formatting that @dot-asm asked for are included.

Reconfirm?

**Comment by richsalz** [comment]
Yes from me, but let @dot-asm look ☺

**Comment by dot-asm** [comment]
Reconfirmed.

**Comment by levitte** [comment]
Merged

**Pull request closed without merging**